ModularTegustation/agent_skills/ranged/grenade_launcher.dm

```diff
@@ -0,0 +1,31 @@
+/obj/effect/proc_holder/ability/aimed/grenade_launcher
+	name = "Grenade Launcher"
+	desc = "Use a gun in hand to fire a grenade. Damage type and amount is based off the gun's damage and type. Costs 10 SP"
+	action_icon_state = "cross_spawn0"
+	base_icon_state = "cross_spawn"
+	cooldown_time = 20 SECONDS
+
+/obj/effect/proc_holder/ability/aimed/grenade_launcher/Perform(target, mob/user)
+	if(ishuman(user))
+		var/mob/living/carbon/human/shooter = user
+		H.adjustSanityLoss(10)
+
+	var/list/gunsinhand = list()
+	for(var/obj/item/ego_weapon/ranged/Gun in shooter.held_items)
+		gunsinhand+=Gun
+	if(!LAZYLEN(gunsinhand))
```

Comment:
use length() instead of LAZYLEN when not working with lazy lists for clarity please
```suggestion
	if(!LAZYLEN(gunsinhand))
```